optout/src/test/java/gov/cms/ab2d/optout/OptOutClientServiceTest.java

```diff
@@ -0,0 +1,124 @@
+package gov.cms.ab2d.optout;
+
+import gov.cms.ab2d.common.model.OptOut;
+import gov.cms.ab2d.common.repository.OptOutRepository;
+import org.apache.http.HttpStatus;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.mockserver.client.MockServerClient;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.matchers.Times;
+import org.mockserver.model.Header;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.MissingResourceException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = SpringBootApp.class)
+class OptOutClientServiceTest {
+    @Autowired
+    private OptOutConverterService cut;
+    @Autowired
+    private OptOutRepository optOutRepository;
+
+    private static int mockServerPort = 8083;
+
+    private static ClientAndServer mockServer;
+
+    private static final String TEST_DIR = "test-data/";
+
+    @BeforeAll
+    public static void setupBFDClient() throws IOException {
+        mockServer = ClientAndServer.startClientAndServer(mockServerPort);
+
+        createMockServerExpectation("/v1/fhir/metadata",
+                200,
+                getRawXML("test-data/meta.xml"),
+                List.of());
+
+        createMockServerExpectation(
+                "/v1/fhir/Patient",
```

Comment:
"/v1/fhir/Patient" is repeated a lot, it could be extracted to a constant.

Reply:
I ended up extracting the whole methods out to reuse